airbyte-ci/connectors/pipelines/pipelines/actions/environments.py

```diff
@@ -365,6 +365,25 @@ def with_python_connector_source(context: ConnectorContext) -> Container:
     return with_python_package(context, testing_environment, connector_source_path)
 
 
+async def apply_python_development_overrides(context: ConnectorContext, connector_container: Container) -> Container:
```

Comment:
I'd love to see an unit test on this. I'll implement one when I'll port it on top of https://github.com/airbytehq/airbyte/pull/30474